catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/CatalogCommands.java

```diff
@@ -87,8 +92,24 @@ protected CatalogFacade getCatalog() throws InterruptedException {
         }
     }
 
+    protected StorageProvider getStorageProvider() {
+        List<StorageProvider> storageProviderList = getAllServices(StorageProvider.class);
+        // The list of catalog providers is sorted by OSGi service ranking, hence should
+        // always set the local catalog provider to the first item in the list.
+        return storageProviderList.get(0);
+    }
+
     protected FilterBuilder getFilterBuilder() throws InterruptedException {
         return getService(FilterBuilder.class);
     }
 
+    protected String getUsernameFromSecuritySubject() {
+        Subject subject = SecurityUtils.getSubject();
+        String name = SubjectUtils.getName(subject, "unknown", true);
+        if (name.contains(",CN=")) {
```

Comment:
Also, checking for `CN=` is not sufficient, you need to check that the principal is correct.

```
SecurityAssertion assertion = principals.oneByType(SecurityAssertion.class);
                if (assertion != null) {
                    Principal principal = assertion.getPrincipal();
                    if (principal instanceof X500Principal) {
```